packages/daemon/src/pet-store.js

```diff
@@ -67,11 +72,27 @@ export const makePetStoreMaker = (filePowers, locator) => {
       return petNames.has(petName);
     };
 
-    /** @param {string} petName */
-    const lookup = petName => {
+    /**
+     * @param {string} petName
+     * @returns {string | undefined}
+     */
+    const identifyLocal = petName => {
       assertValidName(petName);
-      const formulaIdentifier = petNames.get(petName);
-      return formulaIdentifier;
+      return petNames.get(petName);
+    };
+
+    /** @type {import('./types.js').IdentifyFn} */
+    const identify = async maybeNamePath => {
+      const namePath = Array.isArray(maybeNamePath)
+        ? maybeNamePath
+        : [maybeNamePath];
+      const [headName, ...namePathRest] = namePath;
+      const formulaIdentifier = identifyLocal(headName);
+      if (formulaIdentifier === undefined) {
+        return undefined;
+      } else {
+        return identifyFrom(formulaIdentifier, namePathRest);
+      }
```

Comment:
I don’t think this should exist at this level of abstraction but we should factor out an intermediate layer for name hubs between mail and pet store whenever we introduce directories.